### PR TITLE
2021 07 23 elif

### PR DESCRIPTION
**Pull request by SerVB**

### CODE REVIEW

File: python/ast/src/topython/stmt.kt [hunk 1]
```diff
@@ -58,8 +58,19 @@ fun Return.toPython() =
 fun While.toPython() =
     "while ${test.toPython()}:\n${body.toPython().indent()}\n"
 
-fun If.toPython() =
-    "if ${test.toPython()}:\n${(if (body.isNotEmpty()) body.toPython() else "pass").indent()}\n${if (orelse.isNotEmpty()) "else:\n${orelse.toPython().indent()}\n" else ""}"  // todo: for optimization need to support elif
+fun If.toPython(): String {
+    val elsePart = when {
+        orelse.isEmpty() -> ""
+        orelse.size == 1 && orelse.single() is If -> "el${orelse.single().toPython()}"
```
Comment (krzema12) [inline python/ast/src/topython/stmt.kt:64]: `el${orelse.single().toPython()}` -> `elif`? Just simpler :)

Comment (SerVB) [reply python/ast/src/topython/stmt.kt:64]: `${orelse.single().toPython()}` is a full fledged `if condition:<\n>branch1<\n>else:<\n>branch2`. If we just append `el` to it, it becomes a full fledged `elif` ;)

Could you clarify what change you propose?

Comment (krzema12) [reply python/ast/src/topython/stmt.kt:64]: Ah, sorry... Now I see it :) I thought that `${orelse.single().toPython()}` evaluates only to `if`. No change needed!

Comment (SerVB) [reply python/ast/src/topython/stmt.kt:64]: No problem

**Pull request merged**